translation(CSS): web/css/text-overflow (#729)
webdoky/content: files/uk/web/css/text-overflow/index.md

---
title: text-overflow
slug: Web/CSS/text-overflow
page-type: css-property
tags:
  - CSS
  - CSS Property
  - Reference
  - recipe:css-property
browser-compat: css.properties.text-overflow
---

{{CSSRef}}

> Властивість [CSS](/uk/docs/Web/CSS) **`text-overflow`** (переповнення текстом) задає те, як користувачів повідомляють про прихований вміст, що переливається через край. Він може бути обрізаний, може бути показаний еліпс ('`…`') чи якийсь особливий рядок.

Reviewer: Я не зустрічав щоб ellipsis перекладали як еліпс
```suggestion
Властивість [CSS](/uk/docs/Web/CSS) **`text-overflow`** (переповнення текстом) задає те, як користувачів повідомляють про прихований вміст, що переливається через край. Він може бути обрізаний, може бути показана трикрапка ('`…`') чи якийсь особливий рядок.
```

Reviewer: справді, в деяких словниках зустрічається. Також зустрічається пряма калька "еліпсис".
Пропоную еліпс залишити для "ellipse", а для перекладу "ellipsis" використовувати або "еліпсис", або більш специфічний термін "трикрапка" (підозрюю, можуть бути колізії з еліпсисами в мовах штибу китайської, де трикрапка може не мати сенсу)

{{EmbedInteractiveExample("pages/css/text-overflow.html")}}

Властивість `text-overflow` не примушує до жодного переповнення. Щоб змусити текст переливатися через край свого контейнера, треба задати інші властивості CSS: {{cssxref("overflow")}} і {{cssxref("white-space")}}. Наприклад:

```css
overflow: hidden;
white-space: nowrap;
```

Властивість `text-overflow` впливає лише на той вміст, котрий переливається через блоковий контейнер у його _рядковому_ напрямку (не на текст, котрий переливається через нижню рамку, наприклад).

## Синтаксис

Властивість `text-overflow` може бути задана у вигляді одного чи двох значень. Якщо дано одне значення, то воно задає поведінку для кінця строю (правого кінця для тексту зліва направо, лівого кінця для тексту справа наліво). Якщо дані два значення, то перше з них задає поведінку переповнення зліва строю, а друге – справа строю.

```css
text-overflow: clip;
text-overflow: ellipsis ellipsis;
text-overflow: ellipsis " [..]";

/* Глобальні значення */
text-overflow: inherit;
text-overflow: initial;
text-overflow: revert;
text-overflow: revert-layer;
text-overflow: unset;
```

- одне зі значень – ключових слів: `clip`, `ellipsis`, `fade`

> - функція `fade()`, котрій передається {{cssxref("&lt;length&gt;")}} чи {{cssxref("&lt;percentage&gt;")}} для контролю довжини шляху зникнення

Reviewer: ого, я знову дізнався щось нове)

- `<string>`.

### Значення

- `clip` (обрізати)
  - : Усталене значення цієї властивості. Це ключове слово обріже текст на межі [області вмісту](/uk/docs/Web/CSS/CSS_Box_Model/Introduction_to_the_CSS_box_model), таким чином – обрізання може відбутися посередині символу. Щоб обрізати у проміжку між символами, можна задати `text-overflow` зі значенням порожнього рядка, якщо це підтримується цільовими браузерами: `text-overflow: '';`.

> - `ellipsis` (еліпс)

Reviewer: ```suggestion
- `ellipsis` (трикрапка)
```

>   - : Це ключове слово покаже еліпс (`'…'`, `U+2026 HORIZONTAL ELLIPSIS`), що представлятиме обрізаний текст. Еліпс демонструється всередині [області вмісту](/uk/docs/Web/CSS/CSS_Box_Model/Introduction_to_the_CSS_box_model), зменшуючи кількість показаного тексту. Якщо простору для виведення еліпса недостатньо, він обрізається.

Reviewer: Тут треба буде оновити термінологію, якщо ми прийдемо до іншого перекладу

Author: ```suggestion
  - : Це ключове слово покаже трикрапку (`'…'`, `U+2026 HORIZONTAL ELLIPSIS`), що представлятиме обрізаний текст. Трикрапка демонструється всередині [області вмісту](/uk/docs/Web/CSS/CSS_Box_Model/Introduction_to_the_CSS_box_model), зменшуючи кількість показаного тексту. Якщо простору для виведення трикрапки недостатньо, вона обрізається.
```
